src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+import fs from 'fs';
+import commander from 'commander';
+import { PolykeyAgent, PeerInfo } from '../lib/Polykey';
+import {} from '../../tmp/hole-punch-test/publicserver';
+import { actionRunner, pkLogger, PKMessageType, determineNodePath } from '.';
+
+function makeAddPeerCommand() {
+  return new commander.Command('add')
+    .description('add a new peer to the store')
+    .option('-b, --base64 <base64>', 'decode the peer info from a base64 string')
+    .option('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-a, --connected-address <connectedAddress>', 'public address on which the node can be contacted')
+    .option('-r, --relay-key <relayKey>', 'path to the file which contains the public key of the relay peer')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const base64String = options?.base64?.replace('\r', '')?.replace('\n', '');
+
+        let successful: boolean;
+        if (base64String != undefined) {
+          // read in peer info string
+          const { publicKey, connectedAddress, relayPublicKey } = PeerInfo.parseB64(base64String);
+          successful = await client.addPeer(nodePath, publicKey, connectedAddress?.toString() ?? '', relayPublicKey);
+        } else {
+          const publicKeyPath = options.publicKey;
+          // read in publicKey if it exists
+          const publicKey = fs.readFileSync(publicKeyPath).toString();
+
+          const connectedAddress = options.connectedAddress;
+          const relayPublicKeyPath = options.relayKey;
```

Comment:
I don't think you need to keep reassigning things from the `options.*` object. If you aren't changing the data. Just use it directly from `options.*`.

Reply:
yeah that is a fair point, might be able to reduce some lines of code

Reply:
I have removed all of lines that are unnecessarily assigning a `const` but there are some I left in there as they are being used more than once so assigning a `const` reduces code.